packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3227,15 +3243,31 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   }
 
   @override
-  InheritedWidget inheritFromWidgetOfExactType(Type targetType) {
+  InheritedWidget inheritFromElement(InheritedElement ancestor, { Object aspect }) {
+    assert(ancestor != null);
+    assert(() {
+      if (_parent == null) {
+        // We're being deactivated, see deactivateChild()
```

Comment:
can you `assert(_debugLifecycleState == _ElementLifecycle.inactive)` in this case?

Reply:
It's not necessarily inactive (yet) at this point.

The element's _parent is null if InheritFromWidgetOfExactType is called from a State.deactivate() override. FormFieldState does this:

```
  @override
  void deactivate() {
    Form.of(context)?._unregister(this);
    super.deactivate();
  }
```

The `_unregister()` method is trying to remove the field from its form's list of fields.

The deactivate() method is called by deactivateChild() after the child's parent is set to null but before _debugLifecycleState is changed to  _ElementLifecycle.inactive at the end of Element.deactivate():

```
  void deactivate() {
    assert(_debugLifecycleState == _ElementLifecycle.active);
    ...
    assert(() { _debugLifecycleState = _ElementLifecycle.inactive; return true; }());
  }
```

Reply:
I wonder if we could add another lifecycle state for this intermediate mode...